datacube/scripts/dataset.py

```diff
@@ -35,6 +35,31 @@ def maybe_remap(s):
     return maybe_remap
 
 
+def _resolve_uri(uri, doc):
+    loc = doc.location
```

Comment:
I think we should pop the location(s) out of the document, similar to what's done with the lineage, since it's already stored separately in the DB (and is modified separately with the other apis)

Reply:
ok, I will then force it to be a string and not a list, because popping list will loose data as the rest of the code assume single uri when you first inject the dataset into DB